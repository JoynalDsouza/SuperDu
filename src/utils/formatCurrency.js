import {alertError} from './alertError';

export const formatToINR = value => {
  try {
    return new Intl.NumberFormat('en-IN', {
      style: 'currency',
      currency: 'INR',
    }).format(value);
  } catch (e) {
    alertError(e);
    return value;
  }
};
