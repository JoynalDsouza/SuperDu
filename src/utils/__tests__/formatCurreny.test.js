import {formatToINR} from '../formatCurrency';

describe('formatToINR', () => {
  it('take snapshot of formatToINR', () => {
    expect(formatToINR).toMatchSnapshot();
  });
  it('should return a formatted currency', () => {
    const value = 1000;
    expect(formatToINR(value)).toEqual('₹1,000.00');
  });

  it('should return a formatted currency with 0 value', () => {
    const value = 0;
    expect(formatToINR(value)).toEqual('₹0.00');
  });

  it('should return a formatted currency with undefined value', () => {
    const value = undefined;
    expect(formatToINR(value)).toEqual('₹0.00');
  });

  it('should return a formatted currency with null value', () => {
    const value = null;
    expect(formatToINR(value)).toEqual('₹0.00');
  });

  it('should return a formatted currency with string value', () => {
    const value = '1000';
    expect(formatToINR(value)).toEqual('₹1,000.00');
  });

  it('should return a formatted currency with number value', () => {
    const value = 1000;
    expect(formatToINR(value)).toEqual('₹1,000.00');
  });
});
