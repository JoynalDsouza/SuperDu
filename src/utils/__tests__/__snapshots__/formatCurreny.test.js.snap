// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatToINR take snapshot of formatToINR 1`] = `[Function]`;
